homeassistant/components/mqtt/climate.py

```diff
@@ -456,62 +458,65 @@ def handle_climate_attribute_received(
         except ValueError:
             _LOGGER.error("Could not parse %s from %s", template_name, payload)
 
+    @callback
+    def _handle_current_temperature_received(self, msg: ReceiveMessage) -> None:
+        """Handle current temperature coming via MQTT."""
+        self.handle_climate_attribute_received(
+            msg, CONF_CURRENT_TEMP_TEMPLATE, "_attr_current_temperature"
+        )
```

Comment:
If we change the order here so the msg is last, we could make these partials instead and avoid another function call

Reply:
ie

```diff
diff --git a/homeassistant/components/mqtt/climate.py b/homeassistant/components/mqtt/climate.py
index af963215a80..d9e728a4a92 100644
--- a/homeassistant/components/mqtt/climate.py
+++ b/homeassistant/components/mqtt/climate.py
@@ -440,7 +440,7 @@ class MqttTemperatureControlEntity(MqttEntity, ABC):
 
     @callback
     def handle_climate_attribute_received(
-        self, msg: ReceiveMessage, template_name: str, attr: str
+        self, template_name: str, attr: str, msg: ReceiveMessage
     ) -> None:
         """Handle climate attributes coming via MQTT."""
         payload = self.render_template(msg, template_name)
@@ -458,13 +458,6 @@ class MqttTemperatureControlEntity(MqttEntity, ABC):
         except ValueError:
             _LOGGER.error("Could not parse %s from %s", template_name, payload)
 
-    @callback
-    def _handle_current_temperature_received(self, msg: ReceiveMessage) -> None:
-        """Handle current temperature coming via MQTT."""
-        self.handle_climate_attribute_received(
-            msg, CONF_CURRENT_TEMP_TEMPLATE, "_attr_current_temperature"
-        )
-
     @callback
     def _handle_target_temperature_received(self, msg: ReceiveMessage) -> None:
         """Handle target temperature coming via MQTT."""
@@ -497,7 +490,12 @@ class MqttTemperatureControlEntity(MqttEntity, ABC):
         self.add_subscription(
             topics,
             CONF_CURRENT_TEMP_TOPIC,
-            self._handle_current_temperature_received,
+            partial(
+                self.handle_climate_attribute_received,
+                CONF_CURRENT_TEMP_TEMPLATE,
+                "_attr_current_temperature",
+            ),
             {"_attr_current_temperature"},
         )
         self.add_subscription(
```

We should be able to collapse it yet another level as well and get rid of passing the partial to `add_subscription` as `add_subscription` should be able to create it for us from the args so we only have to do it in one place and we only have to pass the template there.  That should avoid a lot of these functions

```diff
@@ -497,7 +490,7 @@ class MqttTemperatureControlEntity(MqttEntity, ABC):
         self.add_subscription(
             topics,
             CONF_CURRENT_TEMP_TOPIC,
-            self._handle_current_temperature_received,
+            CONF_CURRENT_TEMP_TEMPLATE,
             {"_attr_current_temperature"},
         )
         self.add_subscription(
```